composer.json

```diff
@@ -10,20 +10,45 @@
             "homepage": "https://github.com/lmc-eu"
         }
     ],
+    "autoload": {
+        "psr-4": {
+            "Lmc\\CodingStandard\\": "src/"
+        }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "Lmc\\CodingStandard\\": "tests/"
```

Comment:
Why not convetion `"Lmc\\CodingStandard\\Tests\\": "tests/"`?

Reply:
We use the same namespace for Class and its corresponding test.

Reply:
I know :D but why not convention?

Reply:
AFAIK both of the approaches are widely used, so I don't consider any of them as "general convention".

And why it should be in standalone namespace in the first place? I don't see a reason for this.

The same namespace means you don't need to import the class under test namespace, also it makes sense to me to have class and its tests in the same "module" (namespace), because they belongs to each another.

But its still more about personal preference...

Reply:
Similar unconvention as putting src and tests in the same directory, causes in php cs fixer only troubles. 80 % of php-scoper setup took me only this to solve.

It's really painful to work with such packages, where maintainer don't see behing the package common use.

Reply:
Sorry, I don't understand :). This is the way we use it for years and never heard of single issue. And it is not any "hack",  just usual setting of Composer autoloading. But yeah, I can imagine this could cause trouble when developing some low-level stuff.

BTW `src/` and `tests/` in the same (root) directory are used almost everywhere except packages which are then splited to sub-packages (like Symfony). I find most inconvenient when source is not in standalone directory and mixed with `Tests` directory on the same level. Setting code-coverage or phpcs etc. blacklist/whitelist is nightmare in this case.

Reply:
Again, open source code is very different to closed one.

Reply:
True, but I don't think the difference is in the way autoloading is set up :). But this seems like a topic to pub/meetup, so let's don't forget it next time we meet :-D . I'd like to hear about the stuff you deal with :).